packages/x-components/src/components/suggestions/__tests__/highlight.spec.ts

```diff
@@ -0,0 +1,162 @@
+import { mount, Wrapper } from '@vue/test-utils';
+import Highlight from '../../highlight.vue';
+import { getDataTestSelector } from '../../../__tests__/utils';
+
+function renderHighlight({
+  template = '<Highlight v-bind="$attrs"/>',
+  text,
+  highlight,
+  noMatchClass,
+  matchingPartClass,
+  matchClass
+}: RenderHighlightOptions): RenderHighlightAPI {
+  const wrapper = mount(
+    {
+      inheritAttrs: false,
+      components: {
+        Highlight
+      },
+      template
+    },
+    {
+      propsData: {
+        text,
+        highlight,
+        noMatchClass,
+        matchingPartClass,
+        matchClass
+      }
+    }
+  );
+  return {
+    wrapper,
+    getStartPart() {
+      return wrapper.find(getDataTestSelector('highlight-start'));
+    },
+    getMatchingPart() {
+      return wrapper.find(getDataTestSelector('matching-part'));
+    },
+    getEndPart() {
+      return wrapper.find(getDataTestSelector('highlight-end'));
+    },
+    async setHighlight(highlight) {
+      return await wrapper.setProps({ highlight });
+    }
+  };
+}
+
+describe('testing Highlight component', () => {
+  it('highlights the rendered text when the match happens at the beginning', () => {
```

Comment:
In relation to the comment I put in the computed `matchParts` of the Highlight component, could you add a test where the `text` contains spaces at the start or end?